lib/src/rules/avoid_redundant_argument_values.dart

```diff
@@ -80,6 +84,8 @@ class _Visitor extends SimpleAstVisitor {
         final expressionValue = context.evaluateConstant(arg);
         if (expressionValue.value == value) {
           rule.reportLint(arg);
+        } else if (param.isOptionalPositional) {
+          break;
```

Comment:
I think branch this needs to be moved outside the if starting on line 80. Otherwise I think the following case will incorrectly report a lint:
```
void f([int a = 0, int b]) {}
int r = f(0, 1);
```

Reply:
Hmmm.  I'm not sure I'm following.  I've add this test case (and another) and things work as written.

Maybe you can point me to another test I can try?

Reply:
Oh, right. Of course that case worked. I just confused myself while writing the test case.

But I thought we'd decided that we should only create a lint for the last argument, and never for multiple positional arguments. The test case you already had creates lints for multiple arguments, and moving this code would fix that case (assuming it's broken; maybe I was confused about that too).

Reply:
> I thought we'd decided that we should only create a lint for the last argument

We did discuss that but I forgot.

I'll update. 👍